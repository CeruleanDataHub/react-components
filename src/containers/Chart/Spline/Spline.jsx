import React from "react";

import { Chart } from "../Chart";

const options = {
  tooltip: {
    split: true
  }
};

export const Spline = ({ title, xAxis, series }) => (
  <Chart
    title={title}
    xAxis={xAxis}
    series={series}
    options={options}
    type="spline"
  />
);

Spline.propTypes = {
  /** Title of the chart */
  title: Chart.propTypes.title,
  /** Values for X axis */
  xAxis: Chart.propTypes.xAxis,
  /** Data to be graphed */
  series: Chart.propTypes.series
};

Spline.defaultProps = {
  title: Chart.defaultProps.title,
  xAxis: Chart.defaultProps.xAxis,
  series: Chart.defaultProps.series
};
