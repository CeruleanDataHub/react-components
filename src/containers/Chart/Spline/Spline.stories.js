import React from "react";

import { Spline } from "./Spline";

export default {
  title: "Spline",
  component: Spline
};

const data1 = [
  8052,
  2152,
  7085,
  6063,
  9790,
  7649,
  1298,
  8123,
  6219,
  9943,
  9271,
  9072,
  1969,
  3831,
  3241,
  3308,
  3957,
  8083,
  4220,
  5789,
  6752,
  2861,
  4336,
  3286,
  4831,
  7843,
  4559,
  7176,
  6384,
  6083
];

const data2 = [
  1068,
  9641,
  3667,
  9787,
  2471,
  7648,
  5628,
  3380,
  4777,
  1040,
  4612,
  7332,
  4961,
  6425,
  1597,
  4431,
  9885,
  1966,
  3030,
  4524,
  4543,
  9144,
  3796,
  6210,
  6932,
  4136,
  5582,
  9090,
  8860,
  2735
];

const categories = [
  "Jun 26",
  "Jun 27",
  "Jun 28",
  "Jun 29",
  "Jun 30",
  "Jul 1",
  "Jul 2",
  "Jul 3",
  "Jul 4",
  "Jul 5",
  "Jul 6",
  "Jul 7",
  "Jul 8",
  "Jul 9",
  "Jul 10",
  "Jul 11",
  "Jul 12",
  "Jul 13",
  "Jul 14",
  "Jul 15",
  "Jul 16",
  "Jul 17",
  "Jul 18",
  "Jul 19",
  "Jul 20",
  "Jul 21",
  "Jul 22",
  "Jul 23",
  "Jul 24",
  "Jul 25",
  "Jul 26",
  "Jul 27",
  "Jul 28",
  "Jul 29",
  "Jul 30",
  "Jul 31",
  "Aug 1",
  "Aug 2",
  "Aug 3",
  "Aug 4",
  "Aug 5",
  "Aug 6",
  "Aug 7",
  "Aug 8",
  "Aug 9",
  "Aug 10",
  "Aug 11",
  "Aug 12",
  "Aug 13",
  "Aug 14"
];

const xAxis = [{ categories }];

const series = [
  { name: "Temperature", data: data1 },
  { name: "Humidity", data: data2 }
];

export const SplineStory = () => (
  <Spline title="Sauna histogram as spline" xAxis={xAxis} series={series} />
);
