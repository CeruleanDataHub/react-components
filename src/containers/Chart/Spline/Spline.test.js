import { mount } from "enzyme";
import React from "react";

import { Spline } from "./Spline";

describe("Spline", () => {
  it("should render chart", () => {
    const component = mount(<Spline />);

    expect(component.render()).toMatchSnapshot();
  });

  it("should render chart with simple data", () => {
    const xAxis = [
      {
        categories: [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday"
        ]
      }
    ];

    const series = [
      { name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] }
    ];

    const component = mount(
      <Spline title="Chart" xAxis={xAxis} series={series} />
    );

    expect(component.render()).toMatchSnapshot();
  });
});
