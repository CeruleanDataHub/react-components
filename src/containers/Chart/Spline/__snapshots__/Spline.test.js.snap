// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spline should render chart 1`] = `
<Chart
  options={
    Object {
      "tooltip": Object {
        "split": true,
      },
    }
  }
  series={Array []}
  title=""
  type="spline"
  xAxis={null}
/>
`;

exports[`Spline should render chart with simple data 1`] = `
<Chart
  options={
    Object {
      "tooltip": Object {
        "split": true,
      },
    }
  }
  series={
    Array [
      Object {
        "data": Array [
          10,
          15,
          20,
          25,
          15,
          20,
          15,
        ],
        "name": "temperature",
      },
    ]
  }
  title="Chart"
  type="spline"
  xAxis={
    Array [
      Object {
        "categories": Array [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday",
        ],
      },
    ]
  }
/>
`;
